### PR TITLE
Logging and Exception Management

### PR DESCRIPTION
**Pull request by prantaneel-ti**

### CODE REVIEW

File: fast_api_als/utils/adf.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def validate_iso8601(requestdate):
         if match_iso8601(requestdate) is not None:
             return True
     except:
+        log.warning(f'validate_iso8601 is called with requestdate: {requestdate}')
```
Comment (naman-TI) [inline fast_api_als/utils/adf.py]: log.error

File: fast_api_als/services/enrich_lead.py [hunk 2]
```diff
@@ -6,10 +6,26 @@
...
+        log.info('adf_json is not a dict')
+        raise TypeError
+    if adf_json is None or adf_json=={}:
+        #empty dict
+        log.info("adf_json is null or empty")
+        raise ValueError
+    if 'adf' not in adf_json:
+        #field not found
+        log.info("adf field is not found in adf_json object")
+        raise KeyError
```
Comment (naman-TI) [inline fast_api_als/services/enrich_lead.py:25]: look at 'raise' best practices and exception chaining

File: fast_api_als/routers/three_pl.py [hunk 3]
```diff
@@ -10,17 +10,26 @@
...
+        body = json.loads(body)
+        log.info("Body parsed successfully.")
+    except json.JSONDecodeError as jde : 
+        log.error(f'Request body not a valid JSON document: {jde}')
+    except Exception as e:
+        log.error(f'Request body parse failed: {e}')
     provider, role = get_user_role(token)
     if role != "ADMIN" and (role != "3PL"):
+        raise HTTPException(status_code=401, detail='User with role other than ADMIN or 3PL are not authorized.')
         pass
```
Comment (naman-TI) [inline fast_api_als/routers/three_pl.py]: remove line 28, log the error. Line 53 too

File: fast_api_als/routers/three_pl.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import json
 import logging
-from fastapi import Request
+from fastapi import Request, HTTPException
```
Comment (naman-TI) [inline fast_api_als/routers/three_pl.py]: double import

File: fast_api_als/routers/submit_lead.py [hunk 5]
```diff
@@ -70,6 +83,7 @@ async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
     if not validation_check:
         item, path = create_quicksight_data(obj['adf']['prospect'], lead_hash, 'REJECTED', validation_code, {})
         s3_helper_client.put_file(item, path)
```
Comment (naman-TI) [inline fast_api_als/routers/submit_lead.py]: add try-catch line 69,85

**Review by naman-TI — CHANGES_REQUESTED** [review]

File: fast_api_als/utils/adf.py (hunk 1 above)
Comment (prantaneel-ti) [reply fast_api_als/utils/adf.py]: Fixed

File: fast_api_als/routers/three_pl.py (hunk 3 above)
Comment (prantaneel-ti) [reply fast_api_als/routers/three_pl.py]: Fixed.

File: fast_api_als/routers/three_pl.py (hunk 4 above)
Comment (prantaneel-ti) [reply fast_api_als/routers/three_pl.py]: Fixed.

File: fast_api_als/routers/submit_lead.py (hunk 5 above)
Comment (prantaneel-ti) [reply fast_api_als/routers/submit_lead.py]: Fixed.

File: fast_api_als/services/enrich_lead.py (hunk 2 above)
Comment (prantaneel-ti) [reply fast_api_als/services/enrich_lead.py:25]: Fixed.

**Pull request closed without merging**